Tasks/IISWebAppManagementOnMachineGroup/IISWebAppManagementOnMachineGroup.ps1

```diff
@@ -25,42 +25,190 @@ try {
+    $appPoolNameForWebsite = Get-VstsInput -Name "AppPoolNameForWebsite"
+    $dotNetVersionForWebsite = Get-VstsInput -Name "DotNetVersionForWebsite"
+    $pipeLineModeForWebsite = Get-VstsInput -Name "PipeLineModeForWebsite"
+    $appPoolIdentityForWebsite = Get-VstsInput -Name "AppPoolIdentityForWebsite"
+    $appPoolUsernameForWebsite = Get-VstsInput -Name "AppPoolUsernameForWebsite"
+    $appPoolPasswordForWebsite = Get-VstsInput -Name "AppPoolPasswordForWebsite"
+
+    $appCmdCommands = Get-VstsInput -Name "AppCmdCommands"
+
+    $hostName = Get-HostName -protocol $protocol -hostNameWithHttp $hostNameWithHttp -hostNameWithSNI $hostNameWithSNI -hostNameWithOutSNI $hostNameWithOutSNI -sni $serverNameIndication
+
+    Trim-Inputs -siteName ([ref]$websiteName) -physicalPath ([ref]$websitePhysicalPath)  -poolName ([ref]$appPoolNameForWebsite) -physicalPathAuthuser ([ref]$websiteAuthUserName) -appPoolUser ([ref]$appPoolUsernameForWebsite) -sslCertThumbPrint ([ref]$sslCertThumbPrint)
+
+    Validate-Inputs -actionIISWebsite $actionIISWebsite -websiteName $websiteName -createAppPool $createOrUpdateAppPoolForWebsite -appPoolName $appPoolNameForWebsite -addBinding $addBinding -protocol $protocol -sslCertThumbPrint $sslCertThumbPrint
+
+    if ($actionIISWebsite -ieq "CreateOrUpdateWebsite" -and $websitePhysicalPathAuth -ieq "WebsiteWindowsAuth") {
+        $websiteAuthUserPassword = Escape-SpecialChars -str $WebsiteAuthUserPassword
+        $websitePhysicalPathAuthPassword = "$websiteAuthUserPassword" | ConvertTo-SecureString -AsPlainText -Force
+        $websitePhysicalPathAuthCredentials = New-Object System.Management.Automation.PSCredential ("$websiteAuthUserName", $websitePhysicalPathAuthPassword)
+    }
+
+    if ($createOrUpdateAppPoolForWebsite -ieq "true" -and $appPoolIdentityForWebsite -ieq "SpecificUser") {
+        $appPoolPasswordForWebsite = Escape-SpecialChars -str $appPoolPasswordForWebsite
+        $appPoolPasswordForWebsite = "$appPoolPasswordForWebsite" | ConvertTo-SecureString -AsPlainText -Force
+        $appPoolCredentials = New-Object System.Management.Automation.PSCredential ("$appPoolUsernameForWebsite", $appPoolPasswordForWebsite)
+    }
+
+    $appCmdCommands = Escape-SpecialChars -str $appCmdCommands
+
+    Execute-Main -ActionIISWebsite $actionIISWebsite -WebsiteName $websiteName -PhysicalPath $websitePhysicalPath -PhysicalPathAuth $websitePhysicalPathAuth -PhysicalPathAuthCredentials $websitePhysicalPathAuthCredentials -AddBinding $addBinding -Protocol $protocol -IpAddress $ipAddress -Port $port -HostName $hostName -ServerNameIndication $serverNameIndication -SslCertThumbPrint $sslCertThumbPrint -CreateAppPool $createOrUpdateAppPoolForWebsite -AppPoolName $appPoolNameForWebsite -DotNetVersion $dotNetVersionForWebsite -PipeLineMode $pipeLineModeForWebsite -AppPoolIdentity $appPoolIdentityForWebsite -AppPoolCredentials $appPoolCredentials -AppCmdCommands $appCmdCommands
+}
+
+function Get-HostName
```

Comment:
Isn't these helper and part of the PS module? Should we replicate this in all Tasks?